language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -661,22 +661,22 @@ instance ToJSON PackageJson where
 -- Read the provided 'package.json'; transform it to include the
 -- provided workspaces; write it back to disk.
 setupWorkspace :: FilePath -> [(T.Text, [Dependency])] -> FilePath -> IO ()
-setupWorkspace optOutputDir dependencies file = do
+setupWorkspace optOuputDir dependencies file = do
   let (g, nodeFromVertex) = graphFromEdges'
         (map (\(a, ds) -> (a, a, map unDependency ds)) dependencies)
       ps = map (fst3 . nodeFromVertex) $ reverse (topSort g)
         -- Topologically order our packages.
-      scope = scopeOfScopeDir optOutputDir
-        -- 'scope' we expect to be something like "daml2ts".
-  let ourWorkspaces = map ((unScope scope <> "/") <>) ps
+      scopeDir = T.pack $ takeFileName optOuputDir
+        -- The leaf directory of the output directory (e.g. often 'daml2ts').
```

Comment:
It looks like this is the root not the leaf. Am I missing something?

Reply:
No it's the leaf. We take the filename of `/some/long/path/to/here` giving 'here'.

Reply:
Wow. Is it really spelled `optOuputDir` in the code? Better fix that :)

Reply:
Fixed.

Reply:
oh i see, I misread the code. Thanks for the explanation!